rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+                    },
+                    _ => continue,
+                };
+
+                match res {
+                    Ok((current_ans_lookups, ans_lookups)) => {
+                        for (record_key, name_record) in current_ans_lookups {
+                            // Name record may already exist in current_ans_lookup, so need to upsert here
+                            new_current_ans_lookups
+                                .entry(record_key.clone())
+                                .and_modify(|e| {
+                                    e.registered_address = name_record.registered_address.clone();
+                                    // name_record.expiration_timestamp may be zero if primary name wsc occurs before name record wsc,
+                                    // so return the larger of the two timestamps
+                                    e.expiration_timestamp = cmp::max(
+                                        name_record.expiration_timestamp,
+                                        e.expiration_timestamp,
+                                    );
+                                    // name_record.is_primary may be false here, so use OR to preserve is_primary status
+                                    e.is_primary = name_record.is_primary || e.is_primary;
+                                    e.is_deleted = name_record.is_deleted;
+                                })
+                                .or_insert(name_record);
+                        }
+                        for (record_key, name_record) in ans_lookups {
+                            new_ans_lookups
+                                .entry(record_key.clone())
+                                .and_modify(|e| {
+                                    e.registered_address = name_record.registered_address.clone();
+                                    // name_record.expiration_timestamp may be zero if primary name wsc occurs before name record wsc,
+                                    // so return the larger of the two timestamps
+                                    e.expiration_timestamp = cmp::max(
+                                        name_record.expiration_timestamp,
+                                        e.expiration_timestamp,
+                                    );
+                                    // name_record.is_primary may be false here, so use OR to preserve is_primary status
+                                    e.is_primary = name_record.is_primary || e.is_primary;
+                                    e.is_deleted = name_record.is_deleted;
+                                })
+                                .or_insert(name_record);
+                        }
+                    },
+                    Err(e) => {
+                        tracing::error!(
+                            "Failed to parse primary name reverse lookup table item. Error: {:?}",
+                            e
+                        );
```

Comment:
structured error. 

Not sure why you don't panic here but panic in that other function? I'd recommend not panicking anywhere.

Reply:
I think we should panic here though. If it's unable to parse, then the assumption the indexer makes of the underlying data model is wrong and we should fix it.

Reply:
So this requires all the data to be backfilled right? We had a sev about 3 weeks ago where we made this assumnption and a partner didn't backfill so their node broke.

Reply:
Hm ok, if it can't parse, then we can just skip this txn. Also we panic in a lot of other places so we should remove those too.